packages/react/src/auto/hooks/useHasManyInputController.tsx

```diff
@@ -0,0 +1,78 @@
+import { useFieldArray } from "react-hook-form";
+import { uniq } from "../../utils.js";
+import { AutoRelationshipInputProps } from "../interfaces/AutoRelationshipInputProps.js";
+import { useFieldMetadata } from "./useFieldMetadata.js";
+import { useRelatedModelOptions } from "./useRelatedModelOptions.js";
+
+export const useHasManyInputController = (props: AutoRelationshipInputProps) => {
+  const { field } = props;
+  const { path, metadata } = useFieldMetadata(field);
+  const { fields, remove, append } = useFieldArray({ name: path });
+  const { options, pagination, search, selected, relatedModel } = useRelatedModelOptions(props);
+  const removeFromFieldsByRecordId = (recordId: string) => {
+    const index = fields.findIndex((entry) => (entry as any).__id === recordId);
+    if (index > -1) {
+      remove(index);
+    }
+  };
+
+  const onSelectRecord = (recordId: string) => {
+    const isAlreadySelected = selectedRecordIds.includes(recordId);
```

Comment:
tiny nit: its a little confusing that this variable is defined below here instead of above here

Reply:
Updated the order of definitions such that variables come before callbacks